From 00e3896b59a540cb7abd7b4cb4838a451684e7d1 Mon Sep 17 00:00:00 2001
From: Fedor Isakov <fedor.isakov@jetbrains.com>
Date: Thu, 25 Jun 2026 09:54:45 +0200
Subject: [PATCH 08/16] MPS-39881 mps-platform build: package
 mps-platform-provided.zip as build step

The use of TeamCity's mappings for building an artifact
is a bad pattern and makes TC an essential component of the CI.

This change makes builds reproducible and independent of the CI setup.
---
 community/mps/build/build/build.xml | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/community/mps/build/build/build.xml b/community/mps/build/build/build.xml
index c3b012a5cd72f..824b819cbd4a9 100644
--- a/community/mps/build/build/build.xml
+++ b/community/mps/build/build/build.xml
@@ -29,7 +29,7 @@
   </macrodef>
 
   <!-- Build the IJ platform for MPS without custom patches (re-runnable) -->
-  <target name="build" depends="ij-platform,repackage"/>
+  <target name="build" depends="ij-platform,repackage,package-mps-platform-provided"/>
 
   <!-- Code changes required for MPS not yet integrated into the codebase -->
   <target name="patch" description="Apply MPS patches to IDEA Community sources">
@@ -105,4 +105,25 @@
     <delete file="${target.dist}/lib/plugin.xml"/>
   </target>
 
+  <target name="package-mps-platform-provided">
+    <!-- Copied from `IdeaPlatform.buildTypes.PlatformCI#artifactRules` and adapted to ant -->
+    <zip destfile="${target.root}/artifacts/mps-platform-provided.zip" duplicate="preserve">
+      <zipfileset file="${target.root}/artifacts/platform-sources.zip" prefix="lib/src"/>
+      <zipfileset dir="${target.root}/dist.all/lib" prefix="lib"/>
+      <zipfileset dir="${target.root}/dist.all/license" prefix="license"/>
+      <zipfileset dir="${target.root}/dist.all/plugins" prefix="plugins"/>
+      <!--<zipfileset dir="${target.root}/dist.mac.x64/plugins/laf-macos" prefix="plugins/laf-macos"/>-->
+      <!--<zipfileset dir="${target.root}/dist.win.x64/plugins/laf-win10" prefix="plugins/laf-win10"/>-->
+      <zipfileset file="${target.root}/dist.all/build.txt" fullpath="lib/build.txt"/>
+      <zipfileset dir="${target.root}/dist.all/bin" prefix="bin"/>
+      <zipfileset dir="${target.root}/dist.all/build/resources" prefix="build/resources"/>
+      <zipfileset dir="${target.root}/dist.all/build/dependencies" prefix="build/dependencies"/>
+      <zipfileset dir="${target.root}/dist.all/build/tools" prefix="build/tools"/>
+      <zipfileset dir="${target.root}/temp/jna" prefix="lib/jna"/>
+      <zipfileset dir="${target.root}/temp/skiko-awt-runtime-all" prefix="lib/skiko-awt-runtime-all"/>
+      <zipfileset dir="${target.root}/temp/native" prefix="lib/native"/>
+      <zipfileset dir="${target.root}/temp/pty4j" prefix="lib/pty4j"/>
+    </zip>
+  </target>
+  
 </project>
-- 
2.55.0

